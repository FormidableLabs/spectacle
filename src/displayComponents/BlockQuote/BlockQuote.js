import React from 'react';
import PropTypes from 'prop-types';

import {
  StyledBlockQuote,
  StyledBlockQuoteContainer
} from './BlockQuote.style';

function BlockQuote({ children, font, noQuotes, fontSize, color }) {
  return !noQuotes ? (
    <StyledBlockQuoteContainer>
      <StyledBlockQuote font={font} fontSize={fontSize} color={color}>
        &quot;{children}&quot;
      </StyledBlockQuote>
    </StyledBlockQuoteContainer>
  ) : (
    <StyledBlockQuoteContainer>
      <StyledBlockQuote font={font} fontSize={fontSize} color={color}>
        {children}
      </StyledBlockQuote>
    </StyledBlockQuoteContainer>
  );
}

BlockQuote.propTypes = {
  children: PropTypes.node.isRequired,
  color: PropTypes.string,
  font: PropTypes.string,
  fontSize: PropTypes.string,
  noQuotes: PropTypes.bool
};

export { BlockQuote };
