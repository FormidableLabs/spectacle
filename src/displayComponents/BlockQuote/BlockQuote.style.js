import styled, { css } from 'styled-components';

const StyledBlockQuoteContainer = styled.div`
  display: flex;
  height: 100%;
  width: 100%;
  justify-content: center;
  align-items: center;
`;

const StyledBlockQuote = styled.div`
  display: block;
  font-size: ${props => (props.fontSize ? props.fontSize : css`10vw`)};
  color: ${props => (props.color? props.color: props.theme.primaryTextColor)}
  font-family: ${props => (props.font ? props.font : props.theme.font)};
  text-align: center;
  margin: 2rem;
`;

export { StyledBlockQuote, StyledBlockQuoteContainer };
