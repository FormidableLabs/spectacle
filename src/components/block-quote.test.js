import React from 'react';
import { mount } from 'enzyme';
import BlockQuote from './block-quote';

describe('<BlockQuote />', () => {
  test('should render correctly.', () => {
    const context = {
      styles: { components: { blockquote: { fontStyle: 'oblique' } } }
    };
    const wrapper = mount(<BlockQuote>Some Quote</BlockQuote>, { context });
    expect(wrapper).toMatchSnapshot();
  });
});
