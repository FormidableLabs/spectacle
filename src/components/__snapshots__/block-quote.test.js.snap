// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlockQuote /> should render correctly. 1`] = `
<BlockQuote>
  <blockquote
    data-radium={true}
    style={
      Object {
        "fontStyle": "oblique",
      }
    }
  >
    Some Quote
  </blockquote>
</BlockQuote>
`;
